### ISSUE TITLE
Docs: Is metadata required within processor definitions or not? 

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#18689]
### Docs page(s)

https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-processors-for-flat-config
https://eslint.org/docs/latest/extend/custom-processors#custom-processor-specification

### What documentation issue do you want to solve?

The [plugin migration for flat config](https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-processors-for-flat-config) says that for processors, there are no changes required.

However, when reading the [custom processor specification](https://eslint.org/docs/latest/extend/custom-processors#custom-processor-specification) it indicates that we're required to duplicate the plugin's meta data within the processor object. 

### What do you think is the correct solution?

I don't know. Duplicating the metadata seems a little strange. The text says:

> The meta object helps ESLint cache the processor and provide more friendly debug message.

However, it is unclear if that's relating to the top-level meta object, or the meta object within the individual processor definitions.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Good question.

The top-level meta object of the plugin is used when the processor is specified in the format plugin-name/processor-name. For example:

```js
// eslint.config.js
import markdown from "eslint-plugin-markdown";

export default [
    {
        files: ["**/*.md"],
        plugins: {
            markdown
        },
        processor: "markdown/markdown"
    },
    // ... other configs
];
```

Processor's own meta object is used when the processor is specified directly, as a processor object. For example:

```js
// eslint.config.js
import markdown from "eslint-plugin-markdown";

export default [
    {
        files: ["**/*.md"],
        processor: markdown.processors.markdown
    },
    // ... other configs
];
```

It is recommended that both the plugin and each processor provide the meta object, so that features that use meta objects (e.g., `--print-config` and `--cache`) don't depend on the way processor is specified in the configuration.

We should clarify this in the Plugin Migration to Flat Config guide.

**Comment by mdjermanovic** [comment]
@amareshsm feel free to submit a PR and we can discuss the changes there.
